src/types.ts

```diff
@@ -24,35 +24,35 @@ export interface Ctx<P> extends CtxBase {
   match: Match<P>;
 }
 
-export interface AsyncRouteComponentState {
-  Component: AsyncRouteableComponent | null;
+export interface AfterRouteComponentState {
+  Component: AfterRouteableComponent | null;
 }
 
-export interface AsyncComponent {
+export interface AfterComponent {
```

Comment:
Why did you rename this and other types?

Reply:
@stepanh @jaredpalmer I could revert this, I just think `AfterXXXX` is a better name for some of the union types because they might not be async.

For example:

```
export type AfterRouteableComponent<Props = any> =
  | AfterRouteComponentType<RouteComponentProps<Props>>
  | React.ComponentType<RouteComponentProps<Props>>
  | React.ComponentType<Props>;
```

It might not have any async behaviour which makes the prefix confusing IMO.

@jaredpalmer is it a breaking change if it has not been published yet?

Reply:
Makes sense, thanks for the clarification.